### ISSUE TITLE
How to provide our own keys?

### ISSUE THREAD

**Issue by kyrogue** [docker/docker.github.io#6697]
File: [engine/security/trust/trust_key_mng.md](https://docs.docker.com/engine/security/trust/trust_key_mng/)

**Comment by gbarr01** [comment]
Hi, @kyrogue , have you looked at the other docs? 

https://docs.docker.com/engine/security/trust/content_trust/
https://docs.docker.com/engine/security/trust/trust_sandbox/

**Comment by kyrogue** [comment]
Yes, i have, if you look at the below, it tells you it generates, now how do we provide our own?
```
When you push your first tagged image with content trust enabled, the docker client recognizes this is your first push and:

alerts you that it is creating a new root key
requests a passphrase for the root key
generates a root key in the ~/.docker/trust directory
requests a passphrase for the repository key
generates a repository key in the ~/.docker/trust directory
```

**Comment by kyrogue** [comment]
What i have done is use docker trust key generate, and copy the private key generated into CI system, and add the public key of it through `docker trust signer add --key` , however during pushing of a signed image it says no valid signing keys for delegation roles
